app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanel.tsx

```diff
@@ -7,12 +7,28 @@ import { connectionPanelContainer } from './ConnectionPanel.styles';
 import { JenkinsServer } from '../../../../src/common/types';
 import { getAllJenkinsServers } from '../../api/getAllJenkinsServers';
 
-// TODO - add DUPLICATE state once I'm pulling in new data from backend
 export const addConnectedState = (servers: JenkinsServer[]): JenkinsServer[] => {
-	return servers.map((server: JenkinsServer) => ({
-		...server,
-		connectedState: server.pipelines.length === 0 ? ConnectedState.PENDING : ConnectedState.CONNECTED
-	}));
+	const ipAddressSet = new Set<string>();
+
+	return servers
+		.slice() // Create a shallow copy to avoid mutating the original array
+		.sort((a, b) => b.pipelines.length - a.pipelines.length)
```

Comment:
Sorting the servers to ensure that if there are duplicates of the same server, the one with pipelines (if there are any), are prioritised and assigned CONNECTED (otherwise PENDING if no pipelines). Subsequent pipelines will be DUPLICATE.